lib/maybe.mjs

```diff
@@ -0,0 +1,26 @@
+import curry from './curry.mjs'
+
+const maybe = x => {
+  const contents = x
+
+  const isNothing = () =>
+    contents === null || contents === undefined
+
+  const map = fn =>
+    maybe(isNothing() ? null : fn(contents))
+
+  const open = curry(
```

Comment:
`open` seems like the the wrong name here. I don't think I've seen that before. How about `fold`?

Reply:
Yes I was wondering what a good name was for this. Lots of the inspiration for this came from [drboolean's mostly adequate guide](https://drboolean.gitbooks.io/mostly-adequate-guide/ch8.html#schr%C3%B6dingers-maybe), but with more functional programming.

In the guide it uses `Maybe` and `maybe`, which to me is confusing.

Reply:
I think I may need to take a another look at the API. I'm just looking into Elm and how that uses `maybe`

Reply:
Look at my long comment below with Haskell examples. I think the naming conventions will start to make more sense. I can't specifically remember the Dr Boolean example at the moment, but I would say that `Maybe` is a type and `maybe` is a function that acts on a `Maybe`.

Reply:
Ahh yes. Here from your link, lowercase `maybe` is a helper, not implemented inside the `Maybe` constructor. 

> There is, however, an escape hatch. If we would rather return a custom value and continue on, we can use a little helper called maybe.
> ```js
> //  maybe :: b -> (a -> b) -> Maybe a -> b
> var maybe = curry(function(x, f, m) {
>   return m.isNothing() ? x : f(m.__value);
> });